Maven compiler error when running under JDK8

I'm hitting the following error when attempting to run a maven build on my machine using JDK8.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project engine: Fatal error compiling: invalid flag: --release -> [Help 1]
```
This seems to be caused by the use of `<release>8</release>` within the configuration for the compiler plugin, which only works for JDK9+.

Commenting out that line fixes the build error for me, but I'm not sure if that's the right way to go.
I found a [post on StackOverflow](https://stackoverflow.com/a/43103038) mentioning that the `<release>` feature adds a bit more safety to the compilation over just specifying `<source>` and `<target>`.

I should probably just finally upgrade my JDK to 11, but I figure officially stating the desired compiler version will help any future devs who are still on JDK8.

--- a/pom.xml
+++ b/pom.xml
@@ -330,7 +330,6 @@
                     <configuration>
                         <source>1.8</source>
                         <target>1.8</target>
-                        <release>8</release>
                         <failOnError>true</failOnError>
                         <failOnWarning>true</failOnWarning>
                         <compilerArgs>
@@ -538,6 +537,25 @@
                         <artifactId>nexus-staging-maven-plugin</artifactId>
                     </plugin>
                 </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>java-9</id>
+            <activation>
+                <jdk>[9,)</jdk>
+            </activation>
+            <build>
+                <pluginManagement>
+                    <plugins>
+                        <plugin>
+                            <groupId>org.apache.maven.plugins</groupId>
+                            <artifactId>maven-compiler-plugin</artifactId>
+                            <configuration>
+                                <release>8</release>
+                            </configuration>
+                        </plugin>
+                    </plugins>
+                </pluginManagement>
             </build>
         </profile>
     </profiles>

